### ISSUE TITLE
Error converting delays to gains in UVCal

### ISSUE THREAD

**Issue by adampbeardsley** [HERA-Team/pyuvdata#244]
Per @zakiali's comment in Slack:

"I’m getting error in converting delays to gains in UVCal. This error is after the delays have already been converted."

```
In [13]: uv.read_calfits('zen.2457555.42443.xx.HH.uvc.first.calfits')

In [14]: uv.delay_array is None
Out[14]: False

In [15]: uv.gain_array is None
Out[15]: True

In [16]: uv.convert_to_gain()
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
<ipython-input-16-2cf500a391f0> in <module>()
----> 1 uv.convert_to_gain()

/usr/local/lib/python2.7/site-packages/pyuvdata-1.1-py2.7.egg/pyuvdata/uvcal.pyc in convert_to_gain(self, delay_convention, run_check, check_extra, run_check_acceptability)
    530             if run_check:
    531                 self.check(check_extra=check_extra,
--> 532                            run_check_acceptability=run_check_acceptability)
    533         else:
    534             raise(ValueError, 'cal_type is unknown, cannot convert to gain')

/usr/local/lib/python2.7/site-packages/pyuvdata-1.1-py2.7.egg/pyuvdata/uvbase.pyc in check(self, check_extra, run_check_acceptability)
    254                                          'Parameter shape is {pshape}, expected shape is '
    255                                          '{eshape}.'.format(param=p, pshape=np.shape(param.value),
--> 256                                                             eshape=eshape))
    257                     if eshape == ():
    258                         # Single element

ValueError: UVParameter _total_quality_array is not expected shape. Parameter shape is (1, 1, 56, 1), expected shape is (1, 1024, 56, 1).

In [17]: uv.gain_array is None
Out[17]: False

In [18]: uv.delay_array is None
Out[18]: True
```

**Comment by adampbeardsley** [comment]
@zakiali The check to be sure you don't already have a gain array is done at the beginning of the function using the `cal_type` parameter. It looks like you have a UVCal object which has a gain array, but `cal_type` is not set to `'gain'`. Could you check this?
If that's the case, the problem is probably in the converting that was done before.

**Comment by zakiali** [comment]
The cal_type is set to gain after the convert_to_gains. I think it might be that the total_quality_array doesn't get converted to have the size of the gain array.

**Comment by plaplant** [comment]
@zakiali I think you're right. I'm not sure the "right" way to do this conversion, since it's not obvious to me how the chisq values of the delay solutions should be transferred over to the gains solutions. Maybe the right approach is to just set `total_quality_array = None`? If that's the route we want to go, I can make a quick PR to do that.

**Comment by zakiali** [comment]
There are no chisq solutions for the firstcal delay solutions. I just put a default of 1 into them and those are just extended for the quality_array when converting to gains. I think either setting it to None or extending the existing quality would be fine. Maybe @adampbeardsley has some input?

**Comment by adampbeardsley** [comment]
Hmm. I think any solution is probably fine - as long as it's obvious that the values aren't real. So an array of 1's, or maybe -1's should work. Setting the entire array to `None` or even an array of `None`s may not play nice with the checking functions. That's not to say it's not a valid solution, just a little more work to implement.

**Comment by bhazelton** [comment]
When you fix this, @plaplant, can you also add something to the history to make it clear that the gains were converted from delays? And we should also add a test that fails before the fix and passes after.

**Comment by plaplant** [comment]
@bhazelton this was actually fixed in PR #248, but we never closed the issue. However, we don't add anything to the history to note the conversion. Should I make that change as a new PR? Or can we close the issue?

**Comment by bhazelton** [comment]
Let's just close it. I might make the history change as a new PR, but it'll be super simple.
